File: src/main/java/com/kubicz/mavenexecutor/window/CustomCheckBoxList.java
package com.kubicz.mavenexecutor.window;

import java.util.ArrayList;
import java.util.List;

import javax.swing.JCheckBox;
import javax.swing.ListModel;

import com.intellij.ui.CheckBoxList;

public class CustomCheckBoxList extends CheckBoxList<String> {

    public List<String> getSelectedItemNames() {
        List<String> selectedItems = new ArrayList<>();

        ListModel model = getModel();
        if(model == null) {
            return selectedItems;
        }

        for(int i = 0; i < model.getSize(); i++) {
            JCheckBox checkBox = (JCheckBox)model.getElementAt(i);
            if(checkBox.isSelected()) {
                selectedItems.add(checkBox.getText());
            }
        }

        return selectedItems;
    }
}

File: src/main/java/com/kubicz/mavenexecutor/window/CustomCheckBoxList.kt
package com.kubicz.mavenexecutor.window

import java.util.ArrayList

import javax.swing.JCheckBox
import javax.swing.ListModel

import com.intellij.ui.CheckBoxList

class CustomCheckBoxList : CheckBoxList<String>() {

    val selectedItemNames: List<String>
        get() {
            val selectedItems = ArrayList<String>()

            val model = model ?: return selectedItems

            for (i in 0 until model.size) {
                val checkBox = model.getElementAt(i) as JCheckBox
                if (checkBox.isSelected) {
                    selectedItems.add(checkBox.text)
                }
            }

            return selectedItems
        }
}